### PR TITLE
[libc] puts implementation for baremetal

### PR DESCRIPTION
**Pull request by petrhosek**
This is a simple baremetal implementation of puts akin to putchar.

### CODE REVIEW

File: libc/src/stdio/baremetal/puts.cpp [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+#include "src/stdio/puts.h"
+#include "src/__support/CPP/string_view.h"
+#include "src/__support/OSUtil/io.h"
+
+namespace LIBC_NAMESPACE {
+
+LLVM_LIBC_FUNCTION(int, puts, (const char *__restrict str)) {
+  cpp::string_view str_view(str);
+
+  // TODO: Can we combine these to avoid needing two writes?
```
Comment (michaelrj-google) [inline libc/src/stdio/baremetal/puts.cpp:18]: I don't think it's possible to combine these without `malloc`ing, unfortunately.

**Review by michaelrj-google — APPROVED** [review]

File: libc/src/stdio/baremetal/puts.cpp (hunk 1 above)
Comment (petrhosek) [reply libc/src/stdio/baremetal/puts.cpp:18]: The alternative I was thinking about would be to have a `writev` style API where you can provide a vector of strings but I'm not sure if it's worth it.

Comment (michaelrj-google) [reply libc/src/stdio/baremetal/puts.cpp:18]: up to you if you want to implement that

Comment (petrhosek) [reply libc/src/stdio/baremetal/puts.cpp:18]: Not now, maybe as a future improvement.

**Pull request merged**